src/hotspot/share/services/memReporter.cpp

```diff
@@ -842,10 +842,20 @@ void MemDetailDiffReporter::old_virtual_memory_site(const VirtualMemoryAllocatio
 
 void MemDetailDiffReporter::diff_virtual_memory_site(const VirtualMemoryAllocationSite* early,
   const VirtualMemoryAllocationSite* current) const {
-  assert(early->flag() == current->flag() || early->flag() == mtNone,
-    "Expect the same flag, but %s != %s", NMTUtil::flag_to_name(early->flag()),NMTUtil::flag_to_name(current->flag()));
+  if ((early->flag() != current->flag()) || (current->flag() == mtNone) || (early->flag() == mtNone)) {
```

Comment:
Shouldn't this be checking for `!= mtNone` as that is what would trip the original assert?

Shouldn't this checking code be under `ifdef ASSERT`?

Reply:
For simplicity, I usually just take the assert condition, bracket it, and prefix with !. Saves me having to think up the negation.